### PR TITLE
release/mainnet 2017-11-01

### PR DESCRIPTION
**Pull request by deepfire**
The branch is not pretty, and it deserved dis-entangling before PR creation.

..but we also want to record what was deployed.

### CODE REVIEW

File: deployments/cardano-nodes-env-production.nix [hunk 1]
```diff
@@ -15,7 +15,7 @@ let nodeMap = globals.nodeMap; in
       ntp = mkMonitor ntp_monitor;
       cardano_node_simple_process = mkMonitor cardano_node_simple_process_monitor;
       chain_quality = mkMonitor chain_quality_monitor;
-      failed_cherish_loop = mkMonitor failed_cherish_loop_monitor;
+      # failed_cherish_loop = mkMonitor failed_cherish_loop_monitor;
```
Comment (domenkozar) [inline deployments/cardano-nodes-env-production.nix]: This should be uncommented now

File: deployments/cardano-nodes-env-staging.nix [hunk 2]
```diff
@@ -39,7 +39,7 @@ let nodeMap = globals.nodeMap; in
       });
 
       chain_quality = mkMonitor chain_quality_monitor;
-      failed_cherish_loop = mkMonitor failed_cherish_loop_monitor;
+      # failed_cherish_loop = mkMonitor failed_cherish_loop_monitor;
```
Comment (domenkozar) [inline deployments/cardano-nodes-env-staging.nix]: same here

File: staging-testnet.yaml [hunk 3]
```diff
@@ -19,6 +19,7 @@ gen-cmdline: template -c staging-testnet.yaml --environment staging --nixops /ni
 environment: Staging
 topology: topology-staging.yaml
 name: rc-staging
+installer-bucket: update-system-testing
```
Comment (domenkozar) [inline staging-testnet.yaml]: This doesn't belong here, but is part of #183

File: deployments/report-server-bucket-storage.nix [hunk 4]
```diff
@@ -10,7 +10,7 @@
       # mount unsets PATH when running this
       device = "/run/current-system/sw/bin/s3fs#${resources.s3Buckets.report-server-logs.name}";
       fsType = "fuse";
-      options = [ "_netdev" "allow_other" "iam_role" ];
+      options = [ "_netdev" "allow_other" "nonempty" ("iam_role=${resources.iamRoles.report-role.name}") ];
```
Comment (domenkozar) [inline deployments/report-server-bucket-storage.nix]: Do we really need nonemtpy? I would imagine the folder should be empty at the mount time. Probably report-server itself should depend on this mount to finish first. 

cc @cleverca22

Comment (deepfire) [reply deployments/report-server-bucket-storage.nix]: @domenkozar, it's more reliable, I would say -- should anything land there for any reason, the mount will still succeed.

But I'm also agnostic -- whatever you like.

**Comment by domenkozar** [comment]
If there's anything in that folder, it's misconfiguration. We should make sure that report server starts after S3 has been mounted. Then S3 mounting should make sure the folder is empty, otherwise we're missing reports.

**Comment by deepfire** [comment]
@domenkozar, dropped "nonempty".

**Comment by domenkozar** [comment]
https://github.com/input-output-hk/iohk-ops/blob/develop/modules/report-server.nix#L69 already exists, so looks good!

@deepfire last thing is resolving git conflicts :)

**Comment by deepfire** [comment]
@domenkozar, now supplied from conflict-free areas only!

**Pull request merged**